pkg/cmd/label/create.go

```diff
@@ -137,8 +137,7 @@ func createLabel(client *http.Client, repo ghrepo.Interface, opts *createOptions
 		return err
 	}
 	requestBody := bytes.NewReader(requestByte)
-	result := label{}
-	err = apiClient.REST(repo.RepoHost(), "POST", path, requestBody, &result)
+	err = apiClient.REST(repo.RepoHost(), "POST", path, requestBody, nil)
```

Comment:
@williammartin : would this not result in an InvalidUnmarshalError from [`json.Unmarshal()`](https://pkg.go.dev/encoding/json#Unmarshal) within [`pkg/api/rest_client.go`](https://github.com/cli/go-gh/blob/d88d88f917cd4e0bcaa582bac2bb909bd79f9759/pkg/api/rest_client.go#L108-L111)?

https://github.com/cli/cli/blob/e461c89b7dca8af35d10aeda3751d6d156992341/api/client.go#L102-L110

Reply:
I wouldn't expect it to for the following reasons:
 * The tests don't fail
 * Building and running the CLI doesn't fail
 * https://go.dev/play/p/s7E74tD4P2P

https://github.com/cli/go-gh/blob/d88d88f917cd4e0bcaa582bac2bb909bd79f9759/pkg/api/rest_client.go#L108 here we take a pointer to the thing passed in, which should satisfy `json.Unmarshal` AFAIK

Reply:
I think the reason the `DELETE` logic is safe passing in `nil` is because of https://github.com/cli/go-gh/blob/d88d88f917cd4e0bcaa582bac2bb909bd79f9759/pkg/api/rest_client.go#L98-L101 conditional logic that short circuits the REST client code:

```go
// DoWithContext issues a request with type specified by method to the
// specified path with the specified body.
// The response is populated into the response argument.
func (c *RESTClient) DoWithContext(ctx context.Context, method string, path string, body io.Reader, response interface{}) error {
	url := restURL(c.host, path)
	req, err := http.NewRequestWithContext(ctx, method, url, body)
	if err != nil {
		return err
	}

	resp, err := c.client.Do(req)
	if err != nil {
		return err
	}

	success := resp.StatusCode >= 200 && resp.StatusCode < 300
	if !success {
		defer resp.Body.Close()
		return HandleHTTPError(resp)
	}

        // Delete REST calls return 204 No Content like https://docs.github.com/en/enterprise-cloud@latest/rest/issues/labels?apiVersion=2022-11-28#delete-a-label
	if resp.StatusCode == http.StatusNoContent {
		return nil
	}
	defer resp.Body.Close()

	b, err := io.ReadAll(resp.Body)
	if err != nil {
		return err
	}

        // This should throw an error if nil is passed for a non-DELETE method
	err = json.Unmarshal(b, &response)
	if err != nil {
		return err
	}

	return nil
}
```

Reply:
For further confirmation see https://go.dev/play/p/tH1I9k_n-MC and you c an `&` `into` and see that it's happy with `&` and is `InvalidUnmarshalError` without.

Reply:
> I think the reason the DELETE logic is safe passing in nil

Is a great find. I don't think it changes my mind on `nil` being safe though on its own.

All that said, if you want to create something to unmarshal into and get this shipped be my guest.

Reply:
I'm quite confident this works because of the way interface boxing works in Go. If we take this function:

```
func unmarshal(into interface{}) error {
	res := "{}"
	return json.Unmarshal([]byte(res), &into)
}
```

When a `nil` is passed, what's happening is it's being boxed into the `interface{}` type because interfaces in Go are sort of like a tuple of the type and the value: `[T, V]`. Thus, when we take the address of `into`, we're getting a pointer to the tuple. That's why `Unmarshal` doesn't barf, because it's not receiving nil, it's receiving a valid pointer.

Reply:
Going to do a bit more testing manually just to make sure I understand this and the changes being proposed.  Apologies for delays